docs/enhancements/disconnected-iso-ignition-generation.md

```diff
@@ -0,0 +1,167 @@
+```go
+type OVEIgnitionGenerator struct {
+    executer               executer.Executer
+    mirrorRegistriesConfig mirrorregistries.ServiceMirrorRegistriesConfigBuilder
+    installerCache         installercache.InstallerCache
+    versionsHandler        versions.Handler
+    log                    logrus.FieldLogger
+    workDir                string
+}
+```
+
+**Ignition Generation Flow:**
+When `discovery.ign` is requested for an InfraEnv with `disconnected-iso` image type:
+1. Validate that InfraEnv is bound to a cluster (required for version determination)
+2. Retrieve the cluster from the database to get the exact Openshift version
+3. Create temporary directory structure with required assets:
+   - `cluster-manifests/infraenv.yaml` - InfraEnv manifest with user's SSH key
+   - `cluster-manifests/pull-secret.yaml` - User's pull secret for registry access
+   - `mirror/registries.conf` - OVE-specific registry mirror configuration
+4. Fetch the appropriate openshift-install binary matching the cluster version
+5. Execute `openshift-install agent create unconfigured-ignition --interactive`
+6. Return the ignition content
+
+**Key Constraints:**
+- InfraEnv must be bound to a cluster to determine the Openshift version
+- The openshift-install binary version must match the disconnected ISO version exactly
+- Registry configuration includes hardcoded OVE mirror registries:
+  - `quay.io/openshift-release-dev/ocp-release` → `registry.appliance.openshift.com:5000/openshift/release-images`
+  - `quay.io/openshift-release-dev/ocp-v4.0-art-dev` → `registry.appliance.openshift.com:5000/openshift/release`
+
+**Integration Points:**
+- `bareMetalInventory.V2RegisterDisconnectedCluster`: New handler for creating disconnected clusters
+- `bareMetalInventory.registerClusterInternalWithKind`: Modified to support different cluster kinds and initial statuses
+- `bareMetalInventory.V2DownloadInfraEnvFiles`: Modified to detect `disconnected-iso` type and call OVE ignition generation
+- `cluster.RegisterCluster`: Modified to accept status parameter for direct status assignment
+- Temporary directory management: Creates isolated workspace for each generation
+
+## Design Details
+
+### API Changes
+
+**New Endpoint:**
+- **POST** `/v2/clusters/disconnected` - Creates a cluster with `DisconnectedCluster` kind and `unmonitored` status
+  - Accepts a new set of parameters specific to disconnected clusters (`disconnected-cluster-create-params`)
+  - Returns cluster object with kind set to `DisconnectedCluster`
+  - Cluster is created directly in `unmonitored` status without state machine transitions
+
+**New Parameter Type:**
+- `disconnected-cluster-create-params`: A dedicated parameter set for disconnected clusters
+  - Initially includes only the subset of parameters needed for ISO configuration (SSH keys, pull secret, cluster name, OpenShift version)
```

Comment:
The cluster params generally (and in your implementation) don't affect ISO creation. The ssh key and pull secret will likely be set on and taken from the infraenv, the cluster name is not relevant for the ISO, and we enforce that the version is the same on both.